tssc/step_implementers/generate_metadata/semantic_version.py

```diff
@@ -0,0 +1,96 @@
+"""
+Step Implementer for the generate-metadata step for generating a
+semantic version from other metadata.
+"""
+
+from tssc import TSSCFactory
+from tssc import StepImplementer
+from tssc import DefaultSteps
+
+DEFAULT_ARGS = {
+    'release-branch': 'master'
+}
+
+class SemanticVersion(StepImplementer): # pylint: disable=too-few-public-methods 
+    """
+    StepImplementer for the generate-metadata step for SemanticVersion.
+
+    Raises
+    ------
+    ValueError
+        If given pom file does not exist
+        If given pom file does not contain required elements
+    """
+
+    def __init__(self, config, results_file):
+        super().__init__(config, results_file, DEFAULT_ARGS)
+
+    @classmethod
+    def step_name(cls):
+        return DefaultSteps.GENERATE_METADATA
+
+    def _validate_step_config(self, step_config):
+        """
+        Function for implementers to override to do custom step config validation.
+
+        Parameters
+        ----------
+        step_config : dict
+            Step configuration to validate.
+        """
+
+    def _run_step(self, runtime_step_config):
+        app_version = None
+        pre_release = None
+        build = None
+        release_branch = runtime_step_config['release-branch']
+
+        current_step_results = self.current_step_results()
+        if 'app-version' in runtime_step_config:
+            app_version = runtime_step_config['app-version']
+        elif 'app-version' in current_step_results:
+            app_version = current_step_results['app-version']
+
+        if 'pre-release' in runtime_step_config:
+            pre_release = runtime_step_config['pre-release']
+        elif 'pre-release' in current_step_results:
+            pre_release = current_step_results['pre-release']
+
+        if 'build' in runtime_step_config:
+            build = runtime_step_config['build']
+        elif 'build' in current_step_results:
+            build = current_step_results['build']
+
+        if not app_version:
```

Comment:
why not restructure these if nots as an else on the earlier logic?

Reply:
good catch. fixed.